src/test/ui/reachable/expr_match.stderr

```diff
@@ -17,6 +22,11 @@ error: unreachable statement
 LL |     println!("I am dead");
    |     ^^^^^^^^^^^^^^^^^^^^^^
    |
+note: any code following this expression is unreachable
+  --> $DIR/expr_match.rs:18:31
+   |
+LL |     match () { () if false => return, () => return }
```

Comment:
This message is somewhat sub-optimal. The divergence here is really a property of the entire `match` expression. If the other arm is changed to `() => true`, code following the `match` is no longer unreachable. So, it seems a little strange to be pointing at something within an arm.

Is it worth it to add a special case for macros where all arms diverge?

Reply:
I don't think we should further complicate the type checking logic. This seems good enough as-is.

Reply:
I added it before I saw your message. There was already code handling the case where all arms diverge, so adding this was a minor change.